qiskit/transpiler/passes/optimization/split_2q_unitaries.py

```diff
@@ -0,0 +1,49 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2024.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""Splits each two-qubit gate in the `dag` into two single-qubit gates, if possible without error."""
+
+import numpy as np
+from qiskit.circuit.library import UnitaryGate
+from qiskit.dagcircuit import DAGCircuit
+from qiskit.synthesis.two_qubit.local_invariance import two_qubit_local_invariants
+from qiskit.synthesis.two_qubit.two_qubit_decompose import decompose_two_qubit_product_gate
+from qiskit.transpiler import TransformationPass
+
+
+class Split2QUnitaries(TransformationPass):
+    """Splits each two-qubit gate in the `dag` into two single-qubit gates, if possible without error."""
+
+    def run(self, dag: DAGCircuit):
+        """Run the Split2QUnitaries pass on `dag`."""
+        for node in dag.topological_op_nodes():
+            # skip operations without two-qubits and for which we can not determine a potential 1q split
+            if (
+                len(node.cargs) > 0
+                or len(node.qargs) != 2
+                or not (hasattr(node.op, "to_matrix") and hasattr(node.op, "__array__"))
+                or (hasattr(node.op, "is_parameterized") and node.op.is_parameterized())
+            ):
+                continue
+
+            # check if the node can be represented by single-qubit gates
+            nmat = node.op.to_matrix()
+            if np.all(two_qubit_local_invariants(nmat) == [1, 0, 3]):
```

Comment:
In this case it might be faster to do this manually with:

```suggestion
            local_invairants = two_qubit_local_invariants(nmat)
            if local_invariants[0] == 1 and local_invariants[1] == 0 and local_invariants == 3:
```

since there are only 3 floating point values I would expect the numpy overhead from `np.all()` to actually be higher than just doing the comparisons directly. Also do you need any tolerance on the comparisons here?

Reply:
We do not appear to require any tolerance on the comparison here. `DAGOpNode(op=Instruction(name='cp', num_qubits=2, num_clbits=0, params=[7.490140565847857e-07])` would also yield exactly `[1, 0, 3]`